src/rest.rs

```diff
@@ -57,9 +58,26 @@ async fn raw(Path(id): Path<String>, layer: Extension<Layer>) -> Result<String,
     Ok(layer.get_raw(Id::try_from(id.as_str())?).await?)
 }
 
+async fn download(
+    Path((id, extension)): Path<(String, String)>,
+    layer: Extension<Layer>,
+) -> Result<Response<String>, ErrorResponse> {
+    let raw_string = raw(Path(id.clone()), layer.clone()).await?;
+    let filename = format!("{}.{}", id, extension);
+    let content_type = "text; charset=utf-8";
+    let content_disposition = format!("attachment; filename=\"{}\"", filename);
+
+    Ok(Response::builder()
+        .header(header::CONTENT_TYPE, content_type)
+        .header(header::CONTENT_DISPOSITION, content_disposition)
```

Comment:
I suppose it should be possible to use `Response::from_parts` together with `TypedHeaders` to set the headers in a typesafe manner.

Reply:
Tried using `TypedHeaders` but it doesn't seem very ergonomic for Response building. Also, the header for which the Typing is the most important `ContenDisposition` does not support `attachement` (see https://docs.rs/headers/latest/src/headers/common/content_disposition.rs.html#56).

I now validated the inputs and put constraints on the static strings. Let me know if this ok.

Reply:
It's okay, I will think about it myself a bit. But strange that [this old issue](https://github.com/hyperium/headers/issues/8) is still not resolved.